pkg/kubelet/network/hairpin/hairpin.go

```diff
@@ -40,21 +40,26 @@ var (
 	ethtoolOutputRegex = regexp.MustCompile("peer_ifindex: (\\d+)")
 )
 
-func SetUpContainer(containerPid int, containerInterfaceName string) error {
+func SetUpContainerPid(containerPid int, containerInterfaceName string) error {
 	e := exec.New()
-	return setUpContainerInternal(e, containerPid, containerInterfaceName)
+	return setUpContainerInternal(e, containerPid, "", containerInterfaceName)
 }
 
-func setUpContainerInternal(e exec.Interface, containerPid int, containerInterfaceName string) error {
-	hostIfName, err := findPairInterfaceOfContainerInterface(e, containerPid, containerInterfaceName)
+func SetUpContainerPath(netnsPath string, containerInterfaceName string) error {
+	e := exec.New()
+	return setUpContainerInternal(e, -1, netnsPath, containerInterfaceName)
+}
+
+func setUpContainerInternal(e exec.Interface, containerPid int, netnsPath, containerInterfaceName string) error {
+	hostIfName, err := findPairInterfaceOfContainerInterface(e, containerPid, netnsPath, containerInterfaceName)
 	if err != nil {
 		glog.Infof("Unable to find pair interface, setting up all interfaces: %v", err)
 		return setUpAllInterfaces()
 	}
 	return setUpInterface(hostIfName)
 }
 
-func findPairInterfaceOfContainerInterface(e exec.Interface, containerPid int, containerInterfaceName string) (string, error) {
+func findPairInterfaceOfContainerInterface(e exec.Interface, containerPid int, netnsPath, containerInterfaceName string) (string, error) {
```

Comment:
Mabe just separate this function into findPairInterfaceByPid and findPairInterfaceByNetnsPath